src/main/java/service/ProcessInputData.java

```diff
@@ -0,0 +1,47 @@
+package service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitModel;
+import strategy.Strategy;
+import strategy.StrategyImpl;
+import strategy.handlers.Balance;
+import strategy.handlers.Purchase;
+import strategy.handlers.Return;
+import strategy.handlers.Supply;
+
+public class ProcessInputData {
+    private static final int OPERATION_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int AMOUNT_INDEX = 2;
+    private final Map<String, Strategy> operationsWithFruits;
+
+    public ProcessInputData() {
+        operationsWithFruits = new HashMap<>();
+        operationsWithFruits.put("b", new Balance());
+        operationsWithFruits.put("s", new Supply());
+        operationsWithFruits.put("r", new Return());
+        operationsWithFruits.put("p", new Purchase());
+    }
+
+    private FruitModel getFruitModel(String[] line) {
+        FruitModel fruitModel = new FruitModel();
+        fruitModel.setName(line[NAME_INDEX]);
+        fruitModel.setAmount(Integer.valueOf(line[AMOUNT_INDEX]));
+        return fruitModel;
+    }
```

Comment:
This class parses the data and performs the operation, isn't that a violation of the Single Responsibility principle?